keps/sig-instrumentation/4827-component-statusz/README.md

```diff
@@ -0,0 +1,426 @@
+
+#### As support engineer
+
+- I want to access detailed build information to determine if a customer is running a known problematic version of a component
+
+### Risks and Mitigations
+
+1. **No sensitive data exposed**
+    
+  We will apply the  principle of least privilege by granting only the minimum necessary permissions for accessing /statusz, which is primarily a debugging tool for operators and maintainers.
+  /statusz will not be accessible through general discovery mechanisms like system:discovery. This prevents unintended exposure of potentially sensitive debugging information.
+  
+  We'll leverage Kubernetes RBAC to grant access to /statusz only to specific users and service accounts that require it for operational monitoring and debugging. For this, we will use the existing system:monitoring group ([ref](https://github.com/kubernetes/kubernetes/blob/release-1.31/staging/src/k8s.io/apiserver/pkg/authentication/user/user.go#L73)), ensuring that only authorized personnel can access this endpoint.
+  
+1. **Performance impact**
+    
+  We will take care to carefully consider the computational overhead associated with the statusz endpoint to avoid negatively impacting the server's overall performance.
+
+2. **Essential data only**
+    
+  We will prioritize inclusion of only essential diagnostic and monitoring data that aligns with the intended purpose of the z-page.
+
+3. **Premature Dependency on Unstable Format**
+
+  The alpha release will explicitly support plain text format only, making it clear that the output is not intended for parsing or automated monitoring. The feature will be secured behind a feature gate that is disabled by default, ensuring users opt-in consciously. Also, the endpoint will support specifying a version as a query parameter to facilitate future transitions to structured schema changes without breaking existing integrations, once the usage patterns and requirements are better understood
+
+## Design Details
+
+### Data Format and versioning
+
+Initially, the statusz page will exclusively support a plain text format for responses. This endpoint will support specifying a version param, like /statusz?version=2, to ensure future compatibility. Through this, we can evolve the endpoint and introduce changes to the response schema without impacting clients relying on previous versions, ensuring backward compatibility.
+
+### Authz and authn
+
+Access to the endpoint will be limited to members of the `system:monitoring group` ([ref](https://github.com/kubernetes/kubernetes/blob/release-1.31/staging/src/k8s.io/apiserver/pkg/authentication/user/user.go#L73)), which is how paths like /healthz, /livez, /readyz are restricted today. 
+
+### Endpoint Response Format
+
+#### Data format : text
+
+#### Request
+* Method: **GET** 
+* Endpoint: **/statusz**
+* Header: `Content-Type: text/plain`
+* URL params:
+  * version: (Optional) An integer representing the desired response schema version. Defaults to 1.
```

Comment:
For alpha should this default to v1alpha1 or something like that? That would make it similar to how other APIs are versioned and graduated.

Reply:
If the text content-type is explicitly un-schema'ed, we can omit version entirely until we need a breaking change, I think.

Reply:
+100. I would also vote for this. We can always version it if we decide to structure the output.

Reply:
Sounds good, removed URL params from the spec. Also documented that we plan to add support for versioning through URL parameters once a response schema is established in future (under "Data Format and versioning")